crates/client/src/client/types.rs

```diff
@@ -290,3 +297,46 @@ impl TryFrom<SchemaRelayedTransactionStatus> for RelayedTransactionStatus {
         })
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[allow(clippy::large_enum_variant)]
+pub enum TransactionType {
+    Script(Script),
+    Create(Create),
+    Mint(Mint),
+    Upgrade(Upgrade),
+    Upload(Upload),
+    Blob(BlobTx),
+    Unknown,
+}
```

Comment:
What do you think about following idea?=)

```suggestion
pub enum TransactionType {
    Known(fuel_tx::Transaction),
    Unknown,
}
```